packages/mui-base/src/useTransition/useTransition.ts

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { TransitionContext, TransitionContextValue } from './TransitionContext';
+/**
+ *
+ * API:
+ *
+ * - [useTransition API](https://mui.com/base-ui/api/use-transition/)
+ */
+export function useTransition() {
+  const transitionContext = React.useContext(TransitionContext);
+  if (!transitionContext) {
+    throw new Error('Missing transition context');
+  }
+
+  const {
+    registerTransition,
+    requestEnter: requestOpen,
+    onEntering,
+    onEntered,
+    onExiting,
+    onExited,
+    hasExited,
+  } = transitionContext;
+
+  React.useEffect(() => {
+    return registerTransition();
+  }, [registerTransition]);
+
+  return {
+    onEntering,
+    onEntered,
+    onExiting,
+    onExited,
+    requestedEnter: requestOpen,
+    hasExited,
+  };
+}
+
+export function useTransitionableElement(requestEnter: boolean) {
```

Comment:
Based on the usage in Popup.tsx, shouldn't this be a `Transition` component?

Reply:
Yup, could be as well. I wasn't focusing that much on the implementation as I wanted to check how would it feel to use it (from a developer's perspective). Do you have any remarks about that (especially in comparison to our current solution)?